src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyPairGenerator.java

```diff
@@ -151,7 +151,9 @@ public void initialize(int keySize, SecureRandom random) {
         try {
             checkKeySize(keySize, null);
         } catch (InvalidAlgorithmParameterException e) {
-            throw new InvalidParameterException(e.getMessage());
+            InvalidParameterException ipe = new InvalidParameterException(e.getMessage());
+            ipe.initCause(e);
+            throw ipe;
```

Comment:
Or I think this could be written more simply as:

`throw new InvalidParameterException(e.getMessage()).initCause(e);`